lib/puppet/type/group.rb

```diff
@@ -36,6 +36,9 @@ module Puppet
       newvalue(:absent) do
         provider.delete
       end
+
+    defaultto :present
```

Comment:
One thing @suckatrash pointed out is that the `user` type implements this differently

https://github.com/puppetlabs/puppet/blob/b7c7b08413a69fb3d6b8000e4c74ea00270dc26f/lib/puppet/type/user.rb#L81-L87

@jhelwig thoughts on the right way to implement this? Should we include the check if the resource is managed? It feels like that shouldn't be necessary at all....

Reply:
@suckatrash @jhelwig did some digging, and it looks like the `@resource.managed?` block isn't necessary. There's a lot of magic and hand waving around why that is, since this code dates back a long way, so I can't tell you exactly why. We don't really understand why this would be needed at all.

Anyway, this is good as is. All we need from you are those tests we talked about last week. @jhelwig also noticed the indentation here is off. Could you get this new line to line up with the `end` on line 38?

Thanks!